src/com/amazon/corretto/crypto/provider/EdGen.java

```diff
@@ -0,0 +1,38 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.corretto.crypto.provider;
+
+import java.security.KeyPair;
+import java.security.KeyPairGeneratorSpi;
+import java.security.SecureRandom;
+
+class EdGen extends KeyPairGeneratorSpi {
+  /**
+   * Generates a new Ed25519 key and returns a pointer to it.
+   *
+   * @param params a native pointer created by {@link #buildEcParams(int)}
+   * @param checkConsistency Run additional consistency checks on the generated keypair
```

Comment:
Java doc does not line up, this function accepts no parameters.

Reply:
Updated